web/modules/custom/adms_validator/src/AdmsValidator.php

```diff
@@ -32,8 +32,12 @@ public function __construct(Connection $sparql_endpoint) {
   /**
    * {@inheritdoc}
    */
-  public function validateByGraphUri(string $graph_uri): AdmsValidationResult {
-    $query_result = $this->sparqlEndpoint->query(self::validationQuery($graph_uri));
+  public function validateByGraphUri(string $graph_uri, string $query = NULL): AdmsValidationResult {
```

Comment:
I will gladly implement this as it was an original idea of mine as well, just wanted to have your input. 
However, for the query split up, I agree that it is a very clumsy thing to split it like this. But the full query might take too long to run, and by default the timeout of the http client is 10s. I had to split it, because half the times, the importer was timing out during that step.
Preg splitting it, was the easiest way that I could think that would avoid to fork the remote library for our own, or open issues (considering the importance of the epic)